L1Trigger/TrackerDTC/interface/Setup.h

```diff
@@ -81,21 +85,84 @@ namespace trackerDTC {
     GlobalPoint stubPos(const TTStubRef& ttStubRef) const;
     // empty trackerDTC EDProduct
     TTDTC ttDTC() const { return TTDTC(numRegions_, numOverlappingRegions_, numDTCsPerRegion_); }
+    // checks if stub collection is considered forming a reconstructable track 
+    bool reconstructable(const std::vector<TTStubRef>& ttStubRefs) const;
+    // checks if tracking particle is selected for efficiency measurements
+    bool useForAlgEff(const TrackingParticle& tp) const;
+    //
+    bool useForReconstructable(const TrackingParticle& tp) const { return tpSelectorLoose_(tp); }
+    //
+    int layerId(const TTStubRef& ttStubRef) const;
+    //
+    int barrel(const TTStubRef& ttStubRef) const;
+    //
+    int psModule(const TTStubRef& ttStubRef) const;
+    //
+    TTBV layerMap(const std::vector<int>& ints) const;
+    //
+    TTBV layerMap(const TTBV& hitPattern, const std::vector<int>& ints) const;
+    //
+    std::vector<int> layerMap(const TTBV& hitPattern, const TTBV& ttBV) const;
+    //
+    std::vector<int> layerMap(const TTBV& ttBV) const;
+    //
+    double dZ(const TTStubRef& ttStubRef, double cot) const;
+    //
+    double v1(const TTStubRef& ttStubRef, double cot) const;
+    //
+    double dPhi(const TTStubRef& ttStubRef, double inv2R) const;
+    //
```

Comment:
Group v0() and v1() together, and add comment "uncertainty squared in stub phi & z."

Reply:
done